cli/integration-test/integration/completion_test.clj

```diff
@@ -0,0 +1,45 @@
+(ns integration.completion-test
+  (:require
+   [clojure.test :refer [deftest testing]]
+   [integration.fixture :as fixture]
+   [integration.helper :as h]
+   [integration.lsp :as lsp]))
+
+(lsp/clean-after-test)
+
+(deftest completion
+  (lsp/start-process!)
+  (lsp/request! (fixture/initialize-request
+                 {:initializationOptions {:lint-project-files-after-startup? false}
+                  :capabilities
+                  {:textDocument
+                   {:completion
+                    {:contextSupport true
+                     :completionItemKind {:valueSet (vec (range 1 26))}
+                     :completionItem {:snippetSupport true
+                                      :commitCharactersSupport true
+                                      :preselectSupport true
+                                      :documentationFormat ["markdown" "plaintext"]
+                                      :resolveSupport {:properties ["documentation" "detail" "additionalTextEdits"]}}}}}}))
```

Comment:
Aren't we missing `insertTextModeSupport` under `completion completionItem`?

Reply:
[doc](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#completionClientCapabilities)
the relevant part states:
> Indicates which properties a client can resolve lazily on a completion item.

I understand this to mean: "that's the parts you can add to the original item on a completionItem/Resolve call".
The values in here are what Calva sends.

Reply:
Yeah, that part makes sense, sorry I think I was not clear or my comment should be on few lines above.
I meant that we are not sending InsertTextMode, that are the ones client support, at least emacs send that, maybe this could affect the test somehow?

Reply:
it probably should, but I couldn't find any code that takes that client capability into account.

Reply:
Yeah, this is true, doesn't makes sense to add I guess